### ISSUE TITLE
No interfaces show up after driver seems to load successfully.

### ISSUE THREAD

**Issue by davenmccalla** [lwfinger/rtl8812au#23]
I cloned this repository and built the kernel module with no issues. Typed `insmod 8812au.ko` and didn't get any errors. Typed `lsmod` and verified the kernel module loaded. Typed `ifconfig -a` and no interfaces show up. When I tried `cp 8812au.ko /lib/modules/$(uname -r)/kernel/drivers/net/wireless` then `depmod` then `reboot` the module didn't load automatically. Typed `modprobe 8188au` then `ifconfig -a` and still no interfaces show up. Typed `dmesg` and `[   11.659406] usbcore: registered new interface driver rtl8812au`.

**Comment by davenmccalla** [comment]
Also typed `lsusb-v` returned:
```
Bus 001 Device 002: ID 2357:011e
Device Descriptor:
  bLength                18
  bDescriptorType         1
  bcdUSB               2.10
  bDeviceClass            0 (Defined at Interface level)
  bDeviceSubClass         0
  bDeviceProtocol         0
  bMaxPacketSize0        64
  idVendor           0x2357
  idProduct          0x011e
  bcdDevice            2.00
  iManufacturer           1 Realtek
  iProduct                2 802.11ac WLAN Adapter
  iSerial                 3 00e04c000001
  bNumConfigurations      1
  Configuration Descriptor:
    bLength                 9
    bDescriptorType         2
    wTotalLength           60
    bNumInterfaces          1
    bConfigurationValue     1
    iConfiguration          0
    bmAttributes         0xe0
      Self Powered
      Remote Wakeup
    MaxPower              500mA
    Interface Descriptor:
      bLength                 9
      bDescriptorType         4
      bInterfaceNumber        0
      bAlternateSetting       0
      bNumEndpoints           6
      bInterfaceClass       255 Vendor Specific Class
      bInterfaceSubClass    255 Vendor Specific Subclass
      bInterfaceProtocol    255 Vendor Specific Protocol
      iInterface              2 802.11ac WLAN Adapter
      Endpoint Descriptor:
        bLength                 7
        bDescriptorType         5
        bEndpointAddress     0x84  EP 4 IN
        bmAttributes            2
          Transfer Type            Bulk
          Synch Type               None
          Usage Type               Data
        wMaxPacketSize     0x0200  1x 512 bytes
        bInterval               0
      Endpoint Descriptor:
        bLength                 7
        bDescriptorType         5
        bEndpointAddress     0x05  EP 5 OUT
        bmAttributes            2
          Transfer Type            Bulk
          Synch Type               None
          Usage Type               Data
        wMaxPacketSize     0x0200  1x 512 bytes
        bInterval               0
      Endpoint Descriptor:
        bLength                 7
        bDescriptorType         5
        bEndpointAddress     0x06  EP 6 OUT
        bmAttributes            2
          Transfer Type            Bulk
          Synch Type               None
          Usage Type               Data
        wMaxPacketSize     0x0200  1x 512 bytes
        bInterval               0
      Endpoint Descriptor:
        bLength                 7
        bDescriptorType         5
        bEndpointAddress     0x87  EP 7 IN
        bmAttributes            3
          Transfer Type            Interrupt
          Synch Type               None
          Usage Type               Data
        wMaxPacketSize     0x0040  1x 64 bytes
        bInterval               3
      Endpoint Descriptor:
        bLength                 7
        bDescriptorType         5
        bEndpointAddress     0x08  EP 8 OUT
        bmAttributes            2
          Transfer Type            Bulk
          Synch Type               None
          Usage Type               Data
        wMaxPacketSize     0x0200  1x 512 bytes
        bInterval               0
      Endpoint Descriptor:
        bLength                 7
        bDescriptorType         5
        bEndpointAddress     0x09  EP 9 OUT
        bmAttributes            2
          Transfer Type            Bulk
          Synch Type               None
          Usage Type               Data
        wMaxPacketSize     0x0200  1x 512 bytes
        bInterval               0
Binary Object Store Descriptor:
  bLength                 5
  bDescriptorType        15
  wTotalLength           22
  bNumDeviceCaps          2
  USB 2.0 Extension Device Capability:
    bLength                 7
    bDescriptorType        16
    bDevCapabilityType      2
    bmAttributes   0x00000002
      Link Power Management (LPM) Supported
  SuperSpeed USB Device Capability:
    bLength                10
    bDescriptorType        16
    bDevCapabilityType      3
    bmAttributes         0x00
    wSpeedsSupported   0x000e
      Device can operate at Full Speed (12Mbps)
      Device can operate at High Speed (480Mbps)
      Device can operate at SuperSpeed (5Gbps)
    bFunctionalitySupport   1
      Lowest fully-functional device speed is Full Speed (12Mbps)
    bU1DevExitLat          10 micro seconds
    bU2DevExitLat        1023 micro seconds
Device Status:     0x0001
  Self Powered
```

**Comment by lwfinger** [comment]
I do not know what distro you are using, or how you are controlling wireless interfaces. On my openSUSE Tumbleweed using NetworkManager, I get two wireless interfaces. There are 2 because CONFIG_CONCURRENT_MODE is defined in include/autoconf.h.

**Comment by kelebek333** [comment]
> Bus 001 Device 002: ID **2357:011e**

This USBID is not added on this driver source. If @lwfinger add this ID you can use this driver. This USBID already added that source. https://github.com/aircrack-ng/rtl8812au  You can install from that source if you want.

**Comment by lwfinger** [comment]
Sorry that I missed the fact that your USB device was not in the tables. If you do a 'git pull', 'make' and 'sudo make install', then your device will now be recognized.

**Comment by davenmccalla** [comment]
Without taking the adapter apart to look, all the evidence suggests the adapter uses the rtl8811au device.

I tried using the driver at https://github.com/morrownr/8821au and it didn't work either. Then I tried the driver at https://github.com/aircrack-ng/rtl8812au and it worked. It turns out that both repositories incorrectly map 2357:011e to the RTL8821A driver. However the Makefile for the aircrack-ng/rtl8812au has both CONFIG_RTL8812A and CONFIG_RTL8821A set to 'y' so it ends up working, while the morrownr/8821au only has CONFIG_RTL8812A set to 'y' and doesn't work.

Not sure which driver to use now (newer is not always better):
morrownr/8821au (5.9.3.2)
aircrack-ng/rtl8812au (5.6.4.2)
lwfinger/rtl8812au (4.2.2 or 5.2.6)

Thanks for your help on this.
